ocaml/typing/printtyp.ml

```diff
@@ -1636,19 +1636,12 @@ let rec tree_of_type_decl id decl =
   in
   (* The algorithm for setting [lay] here is described as Case (C1) in
      Note [When to print jkind annotations] *)
-  let lay = match ty, unboxed with
+  let jkind_annotation = match ty, unboxed with
     | (Otyp_abstract, _) | (_, true) ->
         (* (C1.1) from the Note corresponds to Otyp_abstract. Anything
-           but the default must be user-written, so we just look in the
-           attributes. Similarly, look in the attributes for (C1.2), the
```

Comment:
The new comment should mention `(C1.2)` as well. Or just "both cases of `(C1)`" or something